iis-wf/iis-wf-preprocessing/src/main/resources/eu/dnetlib/iis/wf/preprocessing/report/oozie_app/workflow.xml

```diff
@@ -37,6 +41,30 @@
             <arg>-Ipartial_reports=${input_partial_reports}</arg>
             <arg>-Oreport=${output_report}</arg>
         </java>
+        <ok to="report_name_generator" />
+        <error to="fail" />
+    </action>
+    
+    <action name="report_name_generator">
+        <java>
+            <main-class>eu.dnetlib.iis.common.report.ReportFileNameGenerator</main-class>
+            <arg>preprocessing</arg>
+            <capture-output />
+        </java>
+        <ok to="copy-report" />
+        <error to="fail" />
+    </action>
+    
+    <action name="copy-report">
+        <distcp xmlns="uri:oozie:distcp-action:0.1">
+            <job-tracker>${jobTracker}</job-tracker>
+            <name-node>${nameNode}</name-node>
+            <prepare>
+                <mkdir path="${nameNode}${reports_external_path}" />
+            </prepare>
+            <arg>${nameNode}${working_dir}/report.json</arg>
+            <arg>${nameNode}${reports_external_path}/${wf:actionData('report_name_generator')['report_filename']}</arg>
```

Comment:
I guess we could simplify filename generation by defining this argument as:
```
<arg>${nameNode}${reports_external_path}/preprocessing_report_${timestamp()}.json</arg>
```
You can find more details on `timestamp()` usage here:
http://stackoverflow.com/questions/21750151/getting-current-time-in-oozie-workflow

Date part is not that nice but it makes whole definition much simpler: we could get rid of `ReportFileNameGenerator` implementation and skip `report_name_generator` action execution.

What do you think?